diff --git a/neo/BolScriptUtility.cs b/neo/BolScriptUtility.cs
index 8322633b1de9a635d906b6864a9a889b07667dcf..1dbf8d1c1d89f8b76157428d6583a72b3d752c07 100644
--- a/neo/BolScriptUtility.cs
+++ b/neo/BolScriptUtility.cs
@@ -1,4 +1,5 @@
 using Neo.Ledger;
+using Neo.Network.P2P.Payloads;
 using Neo.SmartContract;
 using Neo.VM;
 using System.Linq;
@@ -7,12 +8,34 @@ namespace Neo
 {
     public class BolScriptUtility
     {
-        public static byte[] _bolScript = null;
+        public static InvocationTransaction BolToken => new InvocationTransaction
+        {
+            Version = 1,
+            Script = BolScript,
+            Gas = Fixed8.Zero,
+            Attributes = new TransactionAttribute[0],
+            Inputs = new CoinReference[0],
+            Outputs = new TransactionOutput[0],
+            Witnesses = new Witness[0]
+        };
+
+        public static InvocationTransaction BolDeploy => new InvocationTransaction
+        {
+            Version = 1,
+            Script = BolDeployScript,
+            Gas = Fixed8.Zero,
+            Attributes = new TransactionAttribute[0],
+            Inputs = new CoinReference[0],
+            Outputs = new TransactionOutput[0],
+            Witnesses = new Witness[0]
+        };
+
+        private static byte[] s_bolScript = null;
         public static byte[] BolScript
         {
             get
             {
-                if (_bolScript == null)
+                if (s_bolScript == null)
                 {
                     var bolSettings = ProtocolSettings.Default.BolSettings;
                     byte[] script = System.IO.File.ReadAllBytes(bolSettings.Path);
@@ -32,31 +55,56 @@ namespace Neo
                             bolSettings.Email,
                             bolSettings.Description
                         );
-                        _bolScript = sb.ToArray();
+                        s_bolScript = sb.ToArray();
                     }
                 }
-                return _bolScript;
+                return s_bolScript;
             }
         }
 
-        public static byte[] _bolScriptHash = null;
+        private static byte[] s_bolScriptHash = null;
         public static byte[] BolScriptHash
         {
             get
             {
-                if (_bolScriptHash == null)
+                if (s_bolScriptHash == null)
                 {
-                    _bolScriptHash = ProtocolSettings.Default
+                    s_bolScriptHash = ProtocolSettings.Default
                         .BolSettings
                         .ScriptHash
                         .HexToBytes()
                         .Reverse()
                         .ToArray();
                 }
-                return _bolScriptHash;
+                return s_bolScriptHash;
+            }
+        }
+
+        private static byte[] s_bolDeployScript = null;
+        public static byte[] BolDeployScript
+        {
+            get
+            {
+                if(s_bolDeployScript == null)
+                {
+                    using (var sb = new ScriptBuilder())
+                    {
+                        sb.EmitPush(0);
+                        sb.Emit(OpCode.PACK);
+
+                        sb.EmitPush("deploy");
+
+                        sb.EmitAppCall(BolScriptHash, false);
+
+                        s_bolDeployScript = sb.ToArray();
+                    }
+                }
+                return s_bolDeployScript;
             }
         }
-        
+
+        public static string BolScriptHashString => ProtocolSettings.Default.BolSettings.ScriptHash;
+
         public static bool IsBolInvocation(byte[] script)
         {
             var currentScriptEnd = script.Skip(script.Length - 20).ToArray();
