diff --git forkSrcPrefix/neo/SmartContract/ContractParameter.cs forkDstPrefix/neo/SmartContract/ContractParameter.cs
index 5e78005e132ad01b7e5228aa99e6bad0d5bb6dcf..b4f3b74467300927dce977801409515505f39038 100644
--- forkSrcPrefix/neo/SmartContract/ContractParameter.cs
+++ forkDstPrefix/neo/SmartContract/ContractParameter.cs
@@ -229,19 +229,17 @@ namespace Neo.SmartContract
                     {
                         context.Add(parameter);
                         StringBuilder sb = new StringBuilder();
-                        sb.Append('[');
+                        sb.Append('{');
                         foreach (var item in data)
                         {
-                            sb.Append('{');
                             sb.Append(ToString(item.Key, context));
-                            sb.Append(',');
+                            sb.Append(':');
                             sb.Append(ToString(item.Value, context));
-                            sb.Append('}');
                             sb.Append(", ");
                         }
                         if (data.Count > 0)
                             sb.Length -= 2;
-                        sb.Append(']');
+                        sb.Append('}');
                         return sb.ToString();
                     }
                 default:
