diff --git a/neo-cli/Settings.cs b/neo-cli/Settings.cs
index fc4bd908bd53a2dd70f17da41c698c56e13c480d..9cd143539e4ca4b963e38170998736fe0153608b 100644
--- a/neo-cli/Settings.cs
+++ b/neo-cli/Settings.cs
@@ -1,4 +1,4 @@
-﻿using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Configuration;
 using Neo.Network.P2P;
 using System.Net;
 
@@ -16,7 +16,7 @@ namespace Neo
 
         static Settings()
         {
-            IConfigurationSection section = new ConfigurationBuilder().AddJsonFile("config.json").Build().GetSection("ApplicationConfiguration");
+            IConfigurationSection section = new ConfigurationBuilder().AddJsonFile("config.json").AddEnvironmentVariables().Build().GetSection("ApplicationConfiguration");
             Default = new Settings(section);
         }
 
