diff --git a/neo/Ledger/Blockchain.cs b/neo/Ledger/Blockchain.cs
index 6cd22c035f667fc9e15557291046149d14328888..1dc36fa104235f05adea2702349f66a89b964a20 100644
--- a/neo/Ledger/Blockchain.cs
+++ b/neo/Ledger/Blockchain.cs
@@ -67,10 +67,10 @@ namespace Neo.Ledger
 
         public static readonly Block GenesisBlock = new Block
         {
-            PrevHash = UInt256.Zero,
-            Timestamp = (new DateTime(2016, 7, 15, 15, 8, 21, DateTimeKind.Utc)).ToTimestamp(),
+            PrevHash = UInt256.Parse("D3723E2573F2A15A6101CDA6540CAE4523ABB8BB21D3BEB62CBC5D9C8614087D"), //Bol Ruleset Hash
+            Timestamp = (new DateTime(2021, 2, 10, 16, 30, 0, DateTimeKind.Utc)).ToTimestamp(),
             Index = 0,
-            ConsensusData = 2083236893, //向比特币致敬
+            ConsensusData = 7745883560, //Global World Population
             NextConsensus = GetConsensusAddress(StandbyValidators),
             Witness = new Witness
             {
